api/proto/teleport/secreports/v1/secreports_service.proto

```diff
@@ -0,0 +1,293 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.secreports.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/header/v1/resourceheader.proto";
+import "teleport/secreports/v1/secreports.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/secreports/v1;secreportsv1";
+
+// SecReportsService is a service that manages security reports.
+service SecReportsService {
+  // UpsertAuditQuery upsets an audit query.
+  rpc UpsertAuditQuery(UpsertAuditQueryRequest) returns (google.protobuf.Empty);
```

Comment:
We should really be returning the resource on any write; by returning a pbempty we have no way to return the revision, and anything that depends on knowing the value that was written (like terraform) needs a second roundtrip to read the resource again, with no real way to know if it has changed in the meantime.

Reply:
Good point. I will migrate to Upsert call where old value is returned. Thanks.